### ISSUE TITLE
Come up with testing process

### ISSUE THREAD

**Issue by lukas-bednar** [kubevirt/kubevirt-ansible#3]
We need to think how to test our Ansible playbooks.

There were suggestions to go with Vagrant, but feel free to come with other ways.

**Comment by petrkotas** [comment]
Vagrant is actually nice option, it can be automated. However, guys in TLV are working on lago testing setup. I believe Gal is working on it.

What I would like to see, is automated deployment (vagrant, lago) and automated check, with checklist of what is OK and what NOT.

It will be ideal to have Gal join us and work with us on this together, since this way we do not replicate each others steps.

In first step we can do thinks together here, and organize everything in folders. Once the time comes, the project can split to deploy - test - develop.

@fabiand can you please invite Gal to this repository too? Thank You.

**Comment by lukas-bednar** [comment]
Relevant PR #24

**Comment by gbenhaim** [comment]
@lukas-bednar I think that we can add an initial testing suite.
Do we go with Pytest?

**Comment by lukas-bednar** [comment]
@gbenhaim when I submitted this issue, I had in mind that on every new PR the deployment playbook will be executed.
So something like std-ci executing our kubevirt-release-on-openshift-installation following by creating VM would be sufficient as a first step for me.

But today we have bigger matrix what we can do:
* kubevirt-install `mode=release` vs `mode=dev`
* on openshift vs k8s

What exactly do you have in mind when you say `going with pytest` ?

**Comment by gbenhaim** [comment]
- I think that we should start with `mode=release` on openshift. In the following weeks std-ci will allow configuring `sub-stages` that run in parallel, with this feature we can run both openshift and k8s CI per PR.

- Will the tests be written in python with pytest as the testing framework?

**Comment by lukas-bednar** [comment]
@gbenhaim  thumb up for `sub-stages` that sounds good.
Regarding to pytest, what exactly you want to test ? In the context of this repository we need to make sure that in-coming changes doesn't break our playbooks. Do you have something else in mind ?

**Comment by gbenhaim** [comment]
in-coming changes from where?

**Comment by lukas-bednar** [comment]
@gbenhaim from world via PRs

**Comment by gbenhaim** [comment]
- If an incoming PR breaks our CI, we won't merge it.
- If a change in kubevirt breaks our CI, then we will have to decide if it's a bug in kubevirt or in kubevirt ansible.
- I think that we should start with start/shell/stop VM. If those tests will run in the CI, and there is a failure, it will be easy to share the logs with the guys from kubevirt, so I think that this task is pretty urgent,

**Comment by gbenhaim** [comment]
@lukas-bednar thoughts?

**Comment by lukas-bednar** [comment]
@gbenhaim as a start I would extend `automation/run.sh` script to make a REST API call to list kubevirt's pods. That could be enough as first step.
Then we could deploy demo testvm and make sure that it will get up and running.

That is my expectation from this ticket.

We can start with kubevirt on openshift in relase mode, note we should cover other scenarios which we have playbooks for.

If you have anything else in mind please share!

**Comment by fabiand** [comment]
Yes, i think starting a Vm is sufficient for onw.

ideally we can reuse the stock kubevirt functional tests and run them against the origin deployment.

**Comment by fabiand** [comment]
/me things that we could build the tests intoa container to easily run thjem in ther environemts.
